### ISSUE TITLE
no-unused-vars is not right.

### ISSUE THREAD

**Issue by PolanZ** [eslint/eslint#12554]
**Tell us about your environment**

* **ESLint Version:** v6.2.1
* **Node Version:** v10.16.0
* **npm Version:** 6.9.0

**What parser (default, Babel-ESLint, etc.) are you using?**
Babel-ESLint

**Please show your full configuration:**
```js
module.exports = {
  root: true,
  env: {
    node: true
  },
  'extends': [
    'plugin:vue/essential',
    // '@smart/standard'
  ],
  // required to lint *.vue files
  plugins: [
    'vue'
  ],
  rules: {
    // allow async-await
    'max-len': [2, { 'code': 200 }],
    'space-before-function-paren': ['error', {
      'anonymous': 'always',
      'named': 'never',
      'asyncArrow': 'always'
    }],
    'no-multi-spaces': ['error', {
      'ignoreEOLComments': true
    }],
    'no-trailing-spaces': 'off',
    'no-unused-vars': [1, { 'args': 'none' }],
    'no-console': process.env.NODE_ENV === 'production' ? 'error' : 'off',
    'no-debugger': process.env.NODE_ENV === 'production' ? 'error' : 'off'
  },
  parserOptions: {
    parser: 'babel-eslint'
  }
};

```

<details>
<summary>Configuration</summary>

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**
lint

```js
for (const key in sortType) {
  if (sortType[key].status === this.sortTypeStatus) {
    return sortType[key].name;
  }
}
```

**What did you expect to happen?**

'key' is defined but never used.eslint(no-unused-vars)

![image](https://user-images.githubusercontent.com/8565486/68558265-abe6c800-0473-11ea-9f5c-1017517ea12a.png)
![image](https://user-images.githubusercontent.com/8565486/68558272-b1dca900-0473-11ea-97b8-f00988fe3ee0.png)

but! use in function rule is right. see
![image](https://user-images.githubusercontent.com/8565486/68558417-392a1c80-0474-11ea-8615-9a581d420ae7.png)
![image](https://user-images.githubusercontent.com/8565486/68558444-4fd07380-0474-11ea-95fa-e576aa4606c1.png)

not output warning

**Comment by aladdin-add** [comment]
Duplicate of #12117

**Comment by aladdin-add** [comment]
please upgrade `babel-eslint` to resolve the issue.
